pennylane/optimize/adam.py

```diff
@@ -61,49 +62,79 @@ def __init__(self, stepsize=0.01, beta1=0.9, beta2=0.99, eps=1e-8):
         self.sm = None
         self.t = 0
 
-    def apply_grad(self, grad, x):
+    def apply_grad(self, grad, args):
         r"""Update the variables x to take a single optimization step. Flattens and unflattens
         the inputs to maintain nested iterables as the parameters of the optimization.
 
         Args:
-            grad (array): The gradient of the objective
+            grad (tuple(array)): The gradient of the objective
                 function at point :math:`x^{(t)}`: :math:`\nabla f(x^{(t)})`
-            x (array): the current value of the variables :math:`x^{(t)}`
+            args (tuple): the current value of the variables :math:`x^{(t)}`
 
         Returns:
-            array: the new values :math:`x^{(t+1)}`
+            list: the new values :math:`x^{(t+1)}`
         """
-
+        args_new = list(args)
```

Comment:
Is this just to make a copy of `args`? In that case perhaps `args_new = args.copy()` could be used instead.

Reply:
`args` is a tuple, and thus immutable.  I need something mutable to update in the loop, so I cast it to a list.